packages/cactus-plugin-keychain-vault/src/test/typescript/unit/transit-vault.test.ts

```diff
@@ -0,0 +1,188 @@
+import { createHash } from "crypto";
+import test, { Test } from "tape-promise/tape";
+import axios from "axios";
+import { v4 as uuidv4 } from "uuid";
+import {
+  PluginKeychainVault,
+  IPluginKeychainVaultOptions,
+  VaultTransitKeyType,
+} from "../../../main/typescript/public-api";
+import { v4 as internalIpV4 } from "internal-ip";
+
+import {
+  Containers,
+  VaultTestServer,
+  //   K_DEFAULT_VAULT_DEV_ROOT_TOKEN,
+  K_DEFAULT_VAULT_HTTP_PORT,
+} from "@hyperledger/cactus-test-tooling";
+
+test("transit secret engine", async (t: Test) => {
+  const vaultTestContainer = new VaultTestServer({});
+  await vaultTestContainer.start();
+
+  const ci = await Containers.getById(vaultTestContainer.containerId);
+  const vaultIpAddr = await internalIpV4();
+  const hostPort = await Containers.getPublicPort(
+    K_DEFAULT_VAULT_HTTP_PORT,
+    ci,
+  );
+  const vaultHost = `http://${vaultIpAddr}:${hostPort}`;
+  test.onFinish(async () => {
+    await vaultTestContainer.stop();
+    await vaultTestContainer.destroy();
+  });
+  const mountPath = "/transit";
+  const testToken = "myroot";
+  // mount transit secret engine
+  await axios.post(
+    vaultHost + "/v1/sys/mounts" + mountPath,
+    {
+      type: "transit",
+    },
+    {
+      headers: {
+        "X-Vault-Token": testToken,
+      },
+    },
+  );
+
+  const testECP256 = "test-ec-p256";
+  const testECP384 = "test-ec-p384";
+  const testECP521 = "test-ec-p521";
+
+  const options: IPluginKeychainVaultOptions = {
+    logLevel: "DEBUG",
+    keychainId: uuidv4(),
+    endpoint: vaultHost,
+    token: testToken,
+    transitSecretMountPath: mountPath,
+    instanceId: uuidv4(),
+  };
+
+  t.test("newKey", (t: Test) => {
+    const client = new PluginKeychainVault(options);
```

Comment:
@Zzocker Nit: Please name it something other than `client` because once we create the REST API endpoints for the same thing, then the tests will also have to be written here to use the API client object and at that point it would become confusing to see two client variables that are different slightly. My recommendation is to call it either `plugin` or `keychain` or `keychainPlugin` whichever you prefer.